include/aspect/material_model/diffusion_dislocation_plasticity.h

```diff
@@ -0,0 +1,192 @@
+         * Defining a minimum strain rate stabilizes the viscosity calculation,
+         * which involves a division by the strain rate. Units: $1/s$.
+         */
+        double min_strain_rate;
+        int n_limit;
+        double min_visc;
+        double max_visc;
+        double veff_coefficient;
+        double ref_visc;
+
+        double strain_rate_residual_threshold;
+        unsigned int stress_max_iteration_number;
+
+        double thermal_diffusivity;
+        double heat_capacity;
+        double grain_size;
+        double strain_rate_limit;
+
+        /**
+         * From multicomponent material model: From a list of compositional
+         * fields of length N, we come up with an N+1 length list that which
+         * also includes the fraction of ``background mantle''. This list
+         * should sum to one, and is interpreted as volume fractions.  If the
+         * sum of the compositional_fields is greater than one, we assume that
+         * there is no background mantle (i.e., that field value is zero).
+         * Otherwise, the difference between the sum of the compositional
+         * fields and 1.0 is assumed to be the amount of background mantle.
+         */
+        std::vector<double> compute_volume_fractions(
+          const std::vector<double> &compositional_fields) const;
+        std::vector<double> densities;
+        std::vector<double> thermal_expansivities;
+
+        /**
+         * Enumeration for selecting which viscosity averaging scheme to use.
+         * Select between harmonic, arithmetic, geometric, and
+         * maximum_composition. The max composition scheme simply uses the
+         * viscosity of whichever field has the highes volume fraction.
+        * For each quadrature point, averaging is conducted over the
+        * N compositional fields plus the background field.
+         */
+        double composition_averaging_exponent;
+        double composition_averaging_exponent_inverse;
+
+
+        /*double average_value (const std::vector<double> &composition,
+                              const std::vector<double> &parameter_values,
+                              const double &composition_averaging_exponent) const;*/
+
+        void
+        calculate_isostrain_stress_and_stress_strain_derivative ( const std::vector<double> &volume_fractions,
+                                                                  const double &pressure,
+                                                                  const double &temperature,
+                                                                  const double &strain_rate_ii,
+                                                                  std::vector<double> &stress,
+                                                                  std::vector<double> &stress_strain_deriv,
+                                                                  std::vector<double> &stress_pressure_deriv) const;
+
+
+        std::vector<double> yield_stress;
```

Comment:
For this vector and those below, are they all per-composition? If so, maybe add a comment before the first of these variables to explain this.